tests/director/api/automatic_scenario_assignment_test.go

```diff
@@ -190,3 +190,77 @@ func Test_DeleteAutomaticScenarioAssignmentForSelector(t *testing.T) {
 	saveExample(t, req.Query(), "delete automatic scenario assignments for selector")
 
 }
+
+func TestAutomaticScenarioAssignmentsWholeScenario(t *testing.T) {
+	//GIVEN
+	ctx := context.Background()
+	defaultValue := "DEFAULT"
+	scenario := "test"
+
+	scenarios := []string{defaultValue, scenario}
+	scenariosOnlyDefault := []string{defaultValue}
+
+	tenantID := testTenants.GetIDByName(t, "TestWholeScenario")
+	createScenariosLabelDefinitionWithinTenant(t, ctx, tenantID, scenarios)
+	selector := graphql.LabelSelectorInput{Key: "testkey", Value: "testvalue"}
+	assignment := graphql.AutomaticScenarioAssignmentSetInput{ScenarioName: scenario, Selector: &selector}
+
+	createAutomaticScenarioAssignmentInTenant(t, ctx, assignment, tenantID)
+
+	rtmInput := graphql.RuntimeInput{Name: "test-name"}
+	rtm := createRuntimeFromInputWithinTenant(t, ctx, &rtmInput, tenantID)
+	defer unregisterRuntimeWithinTenant(t, rtm.ID, tenantID)
+
+	t.Run("Scenario is set when label matches selector", func(t *testing.T) {
+		setRuntimeLabelWithinTenant(t, ctx, tenantID, rtm.ID, selector.Key, selector.Value)
+		rtmWithScenarios := getRuntimeWithinTenant(t, ctx, rtm.ID, tenantID)
+		assertScenarios(t, rtmWithScenarios.Labels, scenarios)
+	})
+
+	selector2 := graphql.LabelSelectorInput{Key: "newtestkey", Value: "newtestvalue"}
+	t.Run("Scenario is unset when automatic scenario assignment's selector is changed", func(t *testing.T) {
+		assignment.Selector = &selector2
+		deleteAutomaticScenarioAssignmentForScenarioWithinTenant(t, ctx, tenantID, assignment.ScenarioName)
+		createAutomaticScenarioAssignmentInTenant(t, ctx, assignment, tenantID)
+		rtmWithoutScenarios := getRuntimeWithinTenant(t, ctx, rtm.ID, tenantID)
+		assertScenarios(t, rtmWithoutScenarios.Labels, scenariosOnlyDefault)
+	})
+
+	t.Run("Scenario is set back when label matches selector", func(t *testing.T) {
+		setRuntimeLabelWithinTenant(t, ctx, tenantID, rtm.ID, selector2.Key, selector2.Value)
```

Comment:
Please try to add Runtime with label at once. Because you test the case with setRuntimeLabel in the subtest `Scenario is set back when label on runtime matches selector`